src/modules/adaptors/Convex/ConvexAdaptor.sol

```diff
@@ -0,0 +1,268 @@
+        (address lpToken, , , , , ) = booster.poolInfo(pid);
+        ERC20 lpt = ERC20(lpToken);
+        return lpt;
+        // TODO: decide to use above or alternative way to get lpt below
+        // IBaseRewardPool rewardsPool = IBaseRewardPool(rewardsPool);
+        // return ERC(rewardsPool.stakingToken());
+    }
+
+    /**
+     * @notice When positions are added to the Registry, this function can be used in order to figure out
+     *         what assets this adaptor needs to price, and confirm pricing is properly setup.
+     * @param adaptorData see adaptorData info at top of this smart contract
+     * @return assets Underlying assets for Cellar's respective Convex market position
+     * @dev all breakdowns of LPT pricing and its underlying assets are done through the PriceRouter extension (in accordance to PriceRouterv2 architecture)
+     */
+    function assetsUsed(bytes memory adaptorData) public view override returns (ERC20[] memory assets) {
+        assets = new ERC20[](1);
+        assets[0] = assetOf(adaptorData);
+    }
+
+    /**
+     * @notice This adaptor returns collateral, and not debt.
+     * @return whether adaptor returns debt or not
+     */
+    function isDebt() public pure override returns (bool) {
+        return false;
+    }
+
+    //============================================ Strategist Functions ===========================================
+
+    /**
+     * @notice Allows strategists to deposit and stake LPTs into Convex markets via the respective Convex market Booster contract
+     * @param _pid specified pool ID corresponding to LPT convex market
+     * @param _amount amount of LPT to deposit and stake
+     * @param _stake whether or not to stake Convex wrapped LPTs (into respective BaseRewardPool) after depositing LPTs into Convex Market
+     * TODO: stake bool: not sure if we just always set this to true (might be gas intensive), or if we allow this as a param somehow.
+     */
+    function deposit(uint256 _pid, address _baseRewardPool, uint256 _amount, bool _stake) public {
+        _validatePositionIsUsed(_pid, _baseRewardPool); // validate pid representing convex market within respective booster
+        booster.deposit(_pid, _amount, _stake);
+    }
+
+    /**
+     * @notice Allows strategists to withdraw from Convex markets via Booster contract without claiming rewards
+     * NOTE: this adaptor will always unwrap to CRV LPTs if possible. It will not keep the position in convex wrapped LPT position.
+     * TODO: likely removing this functionality since unless gas is so expensive to withdrawFromBaseRewardPoolAsLPT() where we can specify to get w/ rewards or w/o
+     * @param _pid specified pool ID corresponding to LPT convex market
+     * @param _amount amount of cvxLPT to unstake, and withdraw (TODO: can't recall if this unwraps to LPT)
+     */
+    function withdrawFromBoosterNoRewards(uint256 _pid, address _baseRewardPool, uint256 _amount) public {
```

Comment:
I think we always want to unwrap LP tokens

Reply:
Agreed, I've removed the unrelated code that does not unwrap LPTs.